storage/bootstrap/result/result.go

```diff
@@ -90,38 +90,41 @@ func MergedBootstrapResult(i, j BootstrapResult) BootstrapResult {
 
 type shardResult struct {
 	opts   Options
-	blocks map[ident.Hash]DatabaseSeriesBlocks
+	blocks *Map
 }
 
 // NewShardResult creates a new shard result.
 func NewShardResult(capacity int, opts Options) ShardResult {
 	return &shardResult{
-		opts:   opts,
-		blocks: make(map[ident.Hash]DatabaseSeriesBlocks, capacity),
+		opts: opts,
+		blocks: NewMap(MapOptions{
+			InitialSize: capacity,
+			KeyCopyPool: opts.DatabaseBlockOptions().BytesPool().BytesPool(),
```

Comment:
nit: This reads weird, could you rename `opts.DatabaseBlockOptions().BytesPool()` to `opts.DatabaseBlockOptions().CheckedBytesPool()`.

Reply:
That would change every pool option in the project unfortunately.

Reply:
Cool opened https://github.com/m3db/m3db/issues/535 to track